pandas/tests/indexing/test_partial.py

```diff
@@ -170,11 +170,21 @@ def test_partial_setting_mixed_dtype(self):
         with pytest.raises(ValueError, match=msg):
             df.loc[0] = [1, 2, 3]
 
-        # TODO: #15657, these are left as object and not coerced
+    @pytest.mark.parametrize("dtype", [None, "int64", "Int64"])
+    def test_loc_setitem_expanding_empty(self, dtype):
         df = DataFrame(columns=["A", "B"])
-        df.loc[3] = [6, 7]
 
-        exp = DataFrame([[6, 7]], index=[3], columns=["A", "B"], dtype="object")
+        value = [6, 7]
+        if dtype == "int64":
```

Comment:
you actually can just use ``pd.array`` no?

Reply:
thatll give a PandasArray, just looking for an ndarray here